### PR TITLE
Feat: PUSO gov script

### LINKED ISSUE #202: CGP - Add PSO governance proposal script
**Description**

- Add an upgrade script in the deployment repo that will do the following
    - [ ]  Initialise the new stabletoken
    - [ ]  (Legacy) configure the constitution params
    - [ ]  Add the stabletoken to the reserve
    - [ ]  Enable gas payments using the new stabletoken
    - [ ]  Create the new exchange
    - [ ]  Configure the trading limits, circuit breaker and breakers

### PR DESCRIPTION
**Pull request by denviljclarke**
### Description

This PR adds the upgrade scripts for the governance proposals to enable PUSO. Checks have been added, and fork tests have been updated and are green.

**For Reviewers**
For the swap tests to run, you can run the new [RelayerReport script ](https://github.com/mento-protocol/mento-deployment/blob/485a8ea21960cdcb0a6efe7f8e9ce0001dc73326/script/dev/dev-RelayerReport.sol) to relay rates to `SortedOracles` on Alfajores/Baklava. 

### Other changes

- None

### Tested

- Added checks to verify the proposals set the correct configuration values after being simulated
- Added checks to verify that swaps in the newly added pools work as expected

### Related issues

- Fixes #202 

### Backwards compatibility

- N/A

### Documentation

- None

### CODE REVIEW

File: script/upgrades/PUSO/Config.sol [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    config.stableTokenConfig = stableTokenPUSOConfig();
+  }
+
+  /* ==================== Rate Feed Configuration ==================== */
+
+  /**
+   * @dev Returns the configuration for the PHPUSD rate feed.
+   */
+  function PHPUSD_RateFeedConfig() internal pure returns (Config.RateFeed memory rateFeedConfig) {
+    rateFeedConfig.rateFeedID = Config.rateFeedID("PHPUSD");
```
Comment (philbow61) [inline script/upgrades/PUSO/Config.sol:43]:
```suggestion
    rateFeedConfig.rateFeedID = Config.rateFeedID("relayed:PHPUSD");
```

File: script/upgrades/PUSO/Config.sol [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  ) internal view returns (Config.Pool memory poolConfig) {
+    poolConfig = Config.Pool({
+      asset0: contracts.celoRegistry("StableToken"),
+      asset1: contracts.deployed("StableTokenPHPProxy"),
+      isConstantSum: true,
+      spread: FixidityLib.newFixedFraction(3, 1000), // 0.3%, in line with current DT of chainlink feed
+      referenceRateResetFrequency: 5 minutes,
+      minimumReports: 1,
+      stablePoolResetSize: 10_000_000 * 1e18,
+      referenceRateFeedID: Config.rateFeedID("PHPUSD"),
```
Comment (philbow61) [inline script/upgrades/PUSO/Config.sol:68]:
```suggestion
      referenceRateFeedID: Config.rateFeedID("relayed:PHPUSD"),
```

File: broadcast/PUSO-00-Create-Proxies.sol/62320/run-latest.json
Comment (philbow61) [inline broadcast/PUSO-00-Create-Proxies.sol/62320/run-latest.json:1]: With the rateFeed alias not being on Baklava we won't deploy PUSO to Baklava so I think you can remove this one

**Review by philbow61 — CHANGES_REQUESTED** [review]
LGTM 👍 nice work 
please verify the contracts on mainnet and alfajores

**Review by philbow61 — APPROVED** [review]

File: script/upgrades/PUSO/Config.sol [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        timeStep1: 1 days,
+        limit1: 57 * 1_000_000,
+        enabledGlobal: true,
+        limitGlobal: 57 * 5_000_000
+      })
+    });
+
+    if (Chain.isBaklava() || Chain.isAlfajores()) {
+      poolConfig.minimumReports = 1;
+    }
```
Comment (bowd) [inline script/upgrades/PUSO/Config.sol:93]: We can get rid of this

Comment (bowd) [inline script/upgrades/PUSO/Config.sol:93]: We can get rid of this

File: script/upgrades/PUSO/PUSOChecks.swap.sol [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+  function assertApproxEq(uint256 a, uint256 b, uint256 maxDelta) internal view {
+    uint256 delta = a > b ? a - b : b - a;
+
+    if (delta > maxDelta) {
+      console.log("Diff(%s) between amounts is greater than %s", delta, maxDelta);
+    }
+
+    require(delta <= maxDelta, "Values are not approximately equal. See logs for more information.");
+  }
```
Comment (bowd) [inline script/upgrades/PUSO/PUSOChecks.swap.sol:205]: Does the `assertApproxEq` from forge-std/Test.sol not work here?

**Review by bowd — APPROVED** [review]

**Pull request merged**